### PR TITLE
Add site backtest script

### PR DESCRIPTION
**Pull request by Sukh-P**
# Pull Request

## Description

Adding a site PVNet backtest script which has been modified from the [original script](https://github.com/openclimatefix/PVNet/commit/860e5e08d8955241b493f2a4c3095dce05fa7167) to be able to create a backtest for specific PV sites 

## How Has This Been Tested?

This has been tested by the running the script and verifying the outputs 
## Checklist:

- [X] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pvnet/utils.py [hunk 1]
```diff
@@ -69,6 +69,34 @@ def __call__(self, gsp_id: int) -> Location:
         )
 
 
+class SiteLocationLookup:
```
Comment (Sukh-P) [inline pvnet/utils.py:72]: Not added unit tests for this as this is used only in a single place (site backtest script) so far and is similar to the class above for GSP location lookup (not quite similar enough though IMO that it's worth merging them into one class)

File: scripts/backtest_sites.py [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+            ),
+        )
+
+        with torch.no_grad():
+            # Run batch through model to get 0-1 predictions for all sites
+            device_batch = copy_batch_to_device(batch_to_tensor(batch), device)
+            y_normed_site = model(device_batch).detach().cpu().numpy()
+        da_normed_site = preds_to_dataarray(y_normed_site, model, valid_times, ALL_SITE_IDS)
+
+        # TODO fix this step: Multiply normalised forecasts by capacities and clip negatives
```
Comment (peterdudfield) [inline scripts/backtest_sites.py:331]: Whats wrong with it?

File: pvnet/utils.py (hunk 1 above)
Comment (peterdudfield) [reply pvnet/utils.py:72]: would it be easy to add a quick test, just incase? Makes it a bit more reobust for the future

File: scripts/backtest_sites.py (hunk 2 above)
Comment (Sukh-P) [reply scripts/backtest_sites.py:331]: Seemed to be mixing up which capacities were for which site, haven't dug into this fully yet but might just be a site datapipes issues, left the outputs as a normalised output for now which can be converted to MW in post processing

File: scripts/backtest_sites.py [hunk 3]
```diff
@@ -0,0 +1,447 @@
+"""
```
[the remark this answers is not in the document]
Comment (Sukh-P) [reply scripts/backtest_sites.py:1]: Multiple sites, you give it a list of site ids and it will generate outputs for those for the specific period specified

[the remark this answers is not in the document]
Comment (peterdudfield) [reply scripts/backtest_sites.py:1]: could you add that to the description, or something like that

File: scripts/backtest_sites.py (hunk 2 above)
Comment (peterdudfield) [reply scripts/backtest_sites.py:331]: thanks

File: scripts/backtest_sites.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Sukh-P) [reply scripts/backtest_sites.py:1]: Done

**Review by peterdudfield — APPROVED** [review]
Looks good, just remember to get that precommit step working. 
Normally its a linting error

File: pvnet/utils.py (hunk 1 above)
Comment (Sukh-P) [reply pvnet/utils.py:72]: I've now added a parametrised unit test for this
